miniapp/miniapp.cpp

```diff
@@ -51,15 +51,26 @@ int main(int argc, char** argv) {
 
     try {
         std::cout << util::mask_stream(global_policy::id()==0);
-        banner();
-
         // read parameters
         io::cl_options options = io::read_options(argc, argv, global_policy::id()==0);
-        std::cout << options << "\n";
-        std::cout << "\n";
-        std::cout << ":: simulation to " << options.tfinal << " ms in "
-                  << std::ceil(options.tfinal / options.dt) << " steps of "
-                  << options.dt << " ms" << std::endl;
+
+        // If compiled in dry run mode we have to set up the dry run
+        // communicator to simulate the number of ranks that may have been set
+        // as a command line parameter (if not, it is 1 rank by default)
+        if (global_policy::kind() == communication::global_policy_kind::dryrun) {
+            // Dry run mode requires that each rank has the same number of cells.
+            // Here we increase the total number of cells if required to ensure
+            // that this condition is satisfied.
+            auto cells_per_rank = options.cells/options.dry_run_ranks;
+            if (options.cells % options.dry_run_ranks) {
```

Comment:
One way to mimic imbalances is to add performance models into the spike gather part of the communicator. That is where all of the MPI ranks synchronize in our current event-driven model.
However, the simplicity of the dry run implementation comes from the assumption that all ranks have the same number of cells.